azurelinuxagent/common/osutil/suse.py

```diff
@@ -82,6 +82,11 @@ def __init__(self):
         super(SUSEOSUtil, self).__init__()
         self.dhclient_name = 'wickedd-dhcp4'
 
+    def set_hostname(self, hostname):
+        shellutil.run(
```

Comment:
We are deprecating the `run` function in favour of `run_command`. Please use `run_command` and make sure to handle exceptions since this function throws.
An example usage:

```
        try:
            shellutil.run_command(["systemctl", "start", unit_filename])
        except Exception as e:
            raise CGroupsException("Failed to create and start {0}. Error: {1}".format(unit_filename, ustr(e)))
```